### ISSUE TITLE
"heartbeat" signal/command/function

### ISSUE THREAD

**Issue by define-private-public** [gphoto/libgphoto2#306]
Say if I've already established a connection to a camera, is there a function I can to to check that the connection to the camera is still good?  Would `gp_camera_wait_for_event()` solve that issue?

**Comment by msmeissn** [comment]
yes, wait for event would solve this, it would return io errors if the camera is gone

**Comment by define-private-public** [comment]
How threadsafe is communication with a single camera object?  Say if I call that `gp_camera_wait_for_event()`, then in another thread, a few seconds later I call `gp_camera_full()` in another thread.  What will happen?  Or do you not know?

**Comment by define-private-public** [comment]
For the `timeout` parameter if I set it to `0`, will it wait indefinitely (as I've seen with many other timeouts in other libraries)?  Or am I almost guaranteed a `GP_EVENT_TIMEOUT`?

Also, if I do get a `GP_EVENT_TIMEOUT` or a `GP_EVENT_UNKNOWN`, will the function still return `GP_OK`?

**Comment by define-private-public** [comment]
Okay, from some initial testing, I was getting `GP_ERROR_IO_USB_FIND` when I disconnected my camera and was calling `gp_camera_wait_for_event()`.  Is that the only IO error I should check for (if there is a disconnected), or are there others I should test for too?

**Comment by msmeissn** [comment]
I would just handle all errors and only accept GP_OK as "okay".

**Comment by msmeissn** [comment]
timeout will also report GP_OK and listed GP_EVENT_TIMEOUT in the eventcode

**Comment by define-private-public** [comment]
Okay, that seems to be what I'm getting right now, and it seems to be working for me.

If there was a more formalized function for doing this (e.g. `bool gp_camera_check_connection();`) that would be nice.  If you want to turn this ticket into something track that feature request go head.  If not, feel free to close this.

Thanks again.
